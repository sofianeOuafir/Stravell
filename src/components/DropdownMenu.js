import React from "react";

class DropDownMenu extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showMenu: false,
      title: props.title || 'Show menu',
      titleClassName: props.titleClassName || 'dropdown-menu__title',
      contentClassName: props.contentClassName || 'dropdown-menu__content'
    };
  }

  showMenu = e => {
    e.preventDefault();
    this.setState(() => ({ showMenu: true }), () => {
      document.addEventListener('click', this.closeMenu);
    });
  };

  componentWillUnmount () {
    document.removeEventListener('click', this.closeMenu);
  }

  closeMenu = () => {
    this.setState(() => ({ showMenu: false }), () => {
      document.removeEventListener('click', this.closeMenu);
    });
  }

  render() {
    return (
      <div className="dropdown-menu">
        <button className={this.state.titleClassName} onClick={this.showMenu}>{ this.state.title }</button>
        {this.state.showMenu && (
          <div className={this.state.contentClassName}>
            {this.props.children}
          </div>
        )}
      </div>
    );
  }
}

export default DropDownMenu;
