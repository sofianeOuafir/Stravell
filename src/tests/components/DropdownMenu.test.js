import React from 'react';
import { shallow } from 'enzyme';
import DropdownMenu from './../../components/DropdownMenu';

const props = { title: 'Foo', titleClassName: 'foobar', };
let wrapper;
beforeEach(() => {
  wrapper = shallow(<DropdownMenu {...props}><button>Bar</button></DropdownMenu>);
})

test('should render correctly with default values', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should set the component state to be showMenu = true when the main button get clicked', () => {
  expect(wrapper.state('showMenu')).toEqual(false);
  wrapper.find(`.${props.titleClassName}`).simulate('click', { preventDefault: () => {} });
  expect(wrapper.state('showMenu')).toEqual(true);
  expect(wrapper).toMatchSnapshot();
});


