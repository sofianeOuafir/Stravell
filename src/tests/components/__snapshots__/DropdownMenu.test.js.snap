// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default values 1`] = `
<div
  className="dropdown-menu"
>
  <button
    className="foobar"
    onClick={[Function]}
  >
    <IoMdMenu
      className="mr2"
    />
    Foo
  </button>
</div>
`;

exports[`should set the component state to be showMenu = true when the main button get clicked 1`] = `
<div
  className="dropdown-menu"
>
  <button
    className="foobar"
    onClick={[Function]}
  >
    <IoMdMenu
      className="mr2"
    />
    Foo
  </button>
  <div
    className="dropdown-menu__content"
  >
    <button>
      Bar
    </button>
  </div>
</div>
`;
